src/plugins/intel_gpu/src/graph/loop.cpp

```diff
@@ -549,16 +557,24 @@ void loop_inst::preprocess_backedge_memory() {
             if (is_dynamic()) {
                 if (output_prim->outputs_allocated()) {
                     auto internal_output_prim_mem = output_prim->output_memory_ptr();
-                    if (internal_output_prim_mem->get_layout() == initial_mem->get_layout()) {
+                    auto input_prim = body_network->get_primitive(back_edge.to);
+                    auto input_layout = input_prim->_impl_params->get_output_layout();
+
+                    if (internal_output_prim_mem->get_layout() == initial_mem->get_layout() &&
+                        internal_output_prim_mem->get_layout() == input_layout) {
                         backedge_mem = internal_output_prim_mem;
                         body_network->set_input_data(back_edge.to, backedge_mem);
                         GPU_DEBUG_LOG << idx << ") Get backedge_mem(" << backedge_mem
                                     << ") from back_edge.from(" << back_edge.from << ")" << std::endl;
                     } else {
+                        auto update_initial_memory = initial_mem;
+                        if (update_initial_memory->get_layout() != input_layout) {
+                            update_initial_memory = _network.get_engine().reinterpret_buffer(*initial_mem, input_layout);
+                        }
                         // When input layout is changed or backedge_mem is null
                         // because output layout of body network is not calculated yet,
                         // Set backedge_mem to nullptr and update it after first execution.
-                        body_network->set_input_data(back_edge.to, initial_mem);
+                        body_network->set_input_data(back_edge.to, update_initial_memory);
```

Comment:
To me, this is not intuitive updating "initial memory". As from the original intention initial memory is a memory to initialize. Seems that this change is "if initial memory layout is not same as input layout (backedge.to layout, update initial memory". I am not understainding why we need to update "initial" memory.....

@ahnyoung-paul You approved this change, could you answer my question too?

Reply:
The initial_mem of loop is set during first inference. If 2nd inference changes layout, layout of outer input node is updated. But inner input mem may not updated due to reuse memory. For this case, inner input memory layout needs to be updated by changed outer input layout.

I'll add comment on code to explain this.

Reply:
Umm Now I have a different curiousity regarding the behavior for multiple iteration of body network. 
(1) Outer network's input layout node (lets say A) is {1, 10} and allocated memory is {1, 20} due to the previous allocation 
- And the body network's "backedge.to" node's layout is "{1,10}"
(2) At first iteration of body network, current will set input A's memory to backedge.to node. 
This is okay because it is actually initial memory. 
(3) If the body network runs for 2nd iteration, expected behavior is to assign backedge.from to backedge.to. Because it is "backedge". However current change, it still assigns outer input node A's memory.

Reply:
In my thought initial impl of checking "from memory layout and outer input memory layout" to check whether the iteration is first or not, is wrong. We'll need another handle to chech whether to set outer mem or backedge.from mem to the backedge.to mem. @ahnyoung-paul  Could you respond to this matter?

Reply:
@yeonbok, The issued function (create_concat_memory_map) only works on preprocess memory functions that are called before running the inner network when the input layout is updated. So backedge memory is not related to this issue. I think the main matter is that the network input layout is determined by the input memory layout, which might differ from the node's layout by memory reuse. If we have another way to set a layout of the input node in the network, I suggest it. however, I can't find a way to set the layout without modifying the current network method. So I suggest interpreting the memory layout before setting the input layout of the inner network. 
How about discussing it offline?

Reply:
The reshape node as a outer input node of issued loop has updated impl layout but not-updated output mem buffer layout. On previous commit, this unmatched mem buffer layout was updated in loop ops.
But now current commit updates reshape's output mem buffer layout by reinterpret_buffer in update_output_memory() of reshape.